### ISSUE TITLE
Increase quotas for larger tutorials

### ISSUE THREAD

**Issue by mrocklin** [pangeo-data/pangeo#437]
While giving a tutorial at PyData NYC I hit up against resource quotas both for cores

> Insufficient regional quota to satisfy request: resource "CPUS": request requires '1600.0' and is short '1140.0'. project has a quota of '600.0' with '460.0' available.

And for routes

> Insufficient project quota to satisfy request: resource "ROUTES": request requires '300.0' and is short '194.0'. project has a quota of '300.0' with '106.0' available.

If we want to use this system for larger tutorials we'll probably have to address these limitations, and ask Google for increased quotas.

I had around 200 attendees, and my objective was to give each something like 20 cores.  They were asked in the tutorial to tear down and restart their cluster which not everyone will accomplish, and there is a possibility that they will just create two instead, so there is another factor of two.  

Ideally we would have space for around 10,000 cores.

I think that we can handle the routes quota by just using larger VMs.  300 * 64 is around 20,000

cc @jhamman @lila 
also cc @jcrist @martindurant who might want to use use this system again for a tutorial in DC next month

**Comment by yuvipanda** [comment]
@mrocklin You also have 5000 CPUs as limit for pre-emptible nodes (vs 600 for non-preemptible). Does that help? 

I'd recommend overprovisioning the nodes that are running user notebooks if possible. The actual memory usage per node is going to be a fraction of total requests, so setting memory / CPU requests to be less than memory / CPU limits for user notebooks is most likely OK. You can also use data from prometheus to determine the overprovisioning ratio - we do this on mybinder.org. @minrk might have some code that helps figure out this ratio from prometheus.

**Comment by yuvipanda** [comment]
I've also sent a request for CPU quota increase to Google. Will keep y'all updated.

**Comment by minrk** [comment]
Overprovisioning is going to depend a lot on how much your users actually use and/or overlap. I.e. if you have significant amount of time where all attendees are running an exercise that pegs all CPUs, then overprovisioning might not be a great idea. That said, overprovisioning CPU is a whole lot safer than overprovisioning memory.

[This notebook](https://nbviewer.jupyter.org/gist/minrk/55116017ef202843de01906c18541471) informed the current mybinder.org allocation, where we consistently see that most users don't use ~anything. In a dask workshop, I would expect a very different result, but perhaps a similar strategy would be able to inform a good provisioning plan.

**Comment by mrocklin** [comment]
> You also have 5000 CPUs as limit for pre-emptible nodes (vs 600 for non-preemptible). Does that help?

> Overprovisioning is going to depend a lot on how much your users actually use and/or overlap. 

I definitely don't mind overprovisioning CPUs.  I do mind overprovisioning memory.  From what I can tell I actually don't see a memory based quota.  If that's the case then it seems like we want to get some high memory nodes and over-provision CPUs.  I'm also seeing that we peaked out on SSD storage.  I'll probably tone down the worker node pools.

@jhamman any objection if I rip apart some of the node pools you have currently on the binder.pangeo.io cluster?

**Comment by jhamman** [comment]
@mrocklin - please proceed. The node pools have two types of ssds: 1) ssd boot disks and 2) ssd local storage. We can certainly drop (2) given that we never fully configured that and I don't think we're using it. It would be great if we can keep (1) for performance reasons.

**Comment by mrocklin** [comment]
OK, so after https://github.com/mrocklin/pydata-nyc-2018-tutorial/pull/1 I'm able to burst up to well over where we were before.  I asked for 1000 2-core workers and got them pretty easily.  Additionally we're no longer swamping the user node pool which binder.pangeo.io is using for jupyter notebook servers.  I checked the quotas during this process and everything was well under limits.  I suspect that by applying these changes more broadly that we can more effectively scale up.

What is the best way to apply these change more broadly?  Just put them into the cookie cutter?

Scaling down is another issue.  The nodes stayed around afterwards.  I investigated what they were doing and I saw that there was another set of pods that had uniformly allocated themselves across most of my nodes, one each.  There were also a couple of service pods, like `fluentd-gcp-v2.0.17-9vgd5` and `
kube-proxy-gke-binder-worker-pool-b47e0140-5zk7`, that seemed to keep pods alive.  This supports @dsludwig 's approach of using taints.  I'm going to watch these for the next bit to see if they do die down or not.  If they don't die down quickly then I'll force the node pool into a smaller size.
